Examples/javascript/native/example.i

```diff
@@ -15,7 +15,11 @@ int placeholder() { return 0; }
     static SwigV8ReturnValue JavaScript_do_work(const SwigV8Arguments &args) {
         SWIGV8_HANDLESCOPE();
         const int MY_MAGIC_NUMBER = 5;
+#if (V8_MAJOR_VERSION-0) < 4 && (SWIG_V8_VERSION < 0x031903)
         v8::Handle<v8::Value> jsresult =
+#else
+        v8::Local<v8::Value> jsresult =
+#endif
```

Comment:
I'd strongly consider defining `SwigV8Object` typedef or whatever allowing to write this just as
```suggestion
        SwigV8Object<v8::Value>::type jsresult =
```
to avoid repeating this preprocessor condition over and over and over again.

(ideal would be to avoid the `::type` part, but I don't see how to do it without C++11 `using`).

Reply:
What about using `SWIGV8_VALUE` instead of introducing a new typedef?

Reply:
If we have `SWIGV8_VALUE`, we'd also need `SWIGV8_OBJECT` and `SWIGV8_FUNCTIONHANDLER` and maybe more, as we use `v8::Handle<>` or `v8::Local<>` with a few types, not just `v8::Value`. So it seems like a typedef would be more flexible.

But having a few macros would still be better than writing these `#if`s everywhere, so if the typedef solution doesn't work for some reason, doing what you propose would still be an improvement IMO.

Reply:
Somehow I failed to use typedef hence this my current version with macros.